src/ucs/datastruct/ptr_array.h

```diff
@@ -121,12 +135,133 @@ void *ucs_ptr_array_replace(ucs_ptr_array_t *ptr_array, unsigned index, void *ne
                     !__ucs_ptr_array_is_free(_var = (void*)((_ptr_array)->start[_index])))
 
 
+static inline int
+__ucs_ptr_array_is_free_and_get_step_size(ucs_ptr_array_elem_t _elem, uint32_t *size_free)
+{
+    int is_free = ((uintptr_t)(_elem) & UCS_PTR_ARRAY_FLAG_FREE);
+
+    if (is_free) {
+       *size_free = _elem >> UCS_PTR_ARRAY_SIZE_FREE_SHIFT;
+    } else {
+       *size_free = 1;
+    }
+
```

Comment:
i'd add an assert that `ucs_assert(*size_free > 0)`

Reply:
Fixed.